---
identifier: "CVE-2021-37668"
identifiers:
- "CVE-2021-37668"
- "GHSA-2wmv-37vq-52g5"
package_slug: "pypi/tensorflow"
title: "FPE in `tf.raw_ops.UnravelIndex`"
description: "An attacker can cause denial of service in applications serving models
  using `tf.raw_ops.UnravelIndex` by triggering a division by 0:\n\n```python\nimport
  tensorflow as tf\n\ntf.raw_ops.UnravelIndex(indices=-1, dims=[1,0,2])\n```"
date: "2024-11-13"
pubdate: "2021-08-25"
affected_range: "<2.3.4||>=2.4.0,<2.4.3||>=2.5.0,<2.5.1||==2.5.0"
fixed_versions:
- "2.3.4"
- "2.4.3"
- "2.5.1"
affected_versions: "All versions before 2.3.4, all versions starting from 2.4.0 before
  2.4.3, all versions starting from 2.5.0 before 2.5.1, version 2.5.0"
not_impacted: ""
solution: "Upgrade to versions 2.3.4, 2.4.3, 2.5.1 or above."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-37668"
- "https://github.com/advisories/GHSA-2wmv-37vq-52g5"
- "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-2wmv-37vq-52g5"
- "https://github.com/tensorflow/tensorflow/commit/a776040a5e7ebf76eeb7eb923bf1ae417dd4d233"
- "https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-cpu/PYSEC-2021-581.yaml"
- "https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow-gpu/PYSEC-2021-779.yaml"
- "https://github.com/pypa/advisory-database/tree/main/vulns/tensorflow/PYSEC-2021-290.yaml"
- "https://github.com/tensorflow/tensorflow"
cvss_v2: "AV:L/AC:L/Au:N/C:N/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
uuid: "0d9849e1-07bc-4b7f-b920-3d54b4ca76c1"
cwe_ids:
- "CWE-369"
- "CWE-937"
- "CWE-1035"
